src/plugins/visualizations/public/types.ts

```diff
@@ -17,6 +17,38 @@
  * under the License.
  */
 
+import { SavedObject } from '../../../plugins/saved_objects/public';
+import { ISearchSource, AggConfigOptions } from '../../../plugins/data/public';
+import { SerializedVis, Vis, VisParams } from './vis';
+
+export { Vis, SerializedVis, VisParams };
+
+export interface VisualizationController {
+  render(visData: any, visParams: any): Promise<void>;
+  destroy(): void;
+  isLoaded?(): Promise<void> | void;
+}
+
+export interface SavedVisState {
+  type: string;
+  params: VisParams;
+  aggs: AggConfigOptions[];
+}
+
+export interface ISavedVis {
+  id: string;
+  title: string;
+  description?: string;
+  visState: SavedVisState;
+  searchSource?: ISearchSource;
+  uiStateJSON?: string;
+  savedSearchRefName?: string;
+  savedSearchId?: string;
+}
+
+// @ts-ignore-next-line
```

Comment:
I tried removing the `@ts-ignore` here and only explicitly ignoring the empty interface, but it turns out there is another issue that's being suppressed:

```
Interface 'VisSavedObject' cannot simultaneously extend types 'SavedObject' and 'ISavedVis'.
  Named property 'id' of types 'SavedObject' and 'ISavedVis' are not identical. ts(2320)
```